pkg/operator/quorumguardcontroller/quorumguardcontroller.go

```diff
@@ -1,298 +0,0 @@
-	}
-	return factory.New().ResyncEvery(10*time.Minute).WithInformers(
-		kubeInformers.InformersFor(operatorclient.TargetNamespace).Core().V1().Pods().Informer(),
-		kubeInformers.InformersFor("").Core().V1().Nodes().Informer(),
-		operatorClient.Informer(),
-	).WithSync(c.sync).ToController("QuorumGuardController", eventRecorder.WithComponentSuffix("quorum-guard-controller"))
-}
-
-func (c *QuorumGuardController) getTopologyMode() (configv1.TopologyMode, error) {
-	// right now cluster topology cannot change, infrastructure cr is immutable,
-	// so we set it once in order not to run api call each time
-	if c.clusterTopology != "" {
-		klog.V(4).Infof("HA mode is: %s", c.clusterTopology)
-		return c.clusterTopology, nil
-	}
-
-	var err error
-	c.clusterTopology, err = ceohelpers.GetControlPlaneTopology(c.infrastructureLister)
-	if err != nil {
-		klog.Errorf("Failed to get topology mode %w ", err)
-		return "", err
-	}
-
-	klog.Infof("HA mode is: %s", c.clusterTopology)
-
-	return c.clusterTopology, nil
-}
-
-func (c *QuorumGuardController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	err := c.ensureEtcdGuard(ctx, syncCtx.Recorder())
-	if err != nil {
-		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
-			Type:    "QuorumGuardControllerDegraded",
-			Status:  operatorv1.ConditionTrue,
-			Reason:  "Error",
-			Message: err.Error(),
-		}))
-		if updateErr != nil {
-			syncCtx.Recorder().Warning("QuorumGuardControllerUpdatingStatus", updateErr.Error())
-		}
-		return err
-	}
-
-	_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient,
-		v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
-			Type:   "QuorumGuardControllerDegraded",
-			Status: operatorv1.ConditionFalse,
-			Reason: "AsExpected",
-		}))
-	return updateErr
-}
-
-func (c *QuorumGuardController) ensureEtcdGuard(ctx context.Context, recorder events.Recorder) error {
-	haTopologyMode, err := c.getTopologyMode()
-	if err != nil {
-		return err
-	}
-
-	if haTopologyMode != configv1.HighlyAvailableTopologyMode {
-		return nil
-	}
-
-	replicaCount, err := c.getMastersReplicaCount(ctx)
-	if err != nil {
-		return err
-	}
-
-	if err := c.ensureEtcdGuardDeployment(ctx, replicaCount, recorder); err != nil {
-		return err
-	}
-
-	if err := c.ensureEtcdGuardPDB(ctx, recorder); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// ensureEtcdGuardDeployment if etcd quorum guard deployment doesn't exist or was changed - apply it
-func (c *QuorumGuardController) ensureEtcdGuardDeployment(ctx context.Context, replicaCount int32, recorder events.Recorder) error {
-
-	if c.etcdQuorumGuard == nil {
-		c.etcdQuorumGuard = resourceread.ReadDeploymentV1OrDie(etcd_assets.MustAsset("etcd/quorumguard-deployment.yaml"))
-	}
-	c.etcdQuorumGuard.Spec.Replicas = &replicaCount
-	// use image from release payload
-	c.etcdQuorumGuard.Spec.Template.Spec.Containers[0].Image = c.cliImagePullSpec
```

Comment:
not sure I saw where this is covered in new controller.

Reply:
I think this would require a change to targetconfig controller?

Reply:
This will be handled via [change](https://github.com/openshift/library-go/pull/1095/commits/787aada078840739db123df54b42875813a02066#diff-73715cadb7bef6b70b2bf5114fb65487380c78bd9a6605b93191fab69e9c46a1R31)